lithium/lithium.py

```diff
@@ -55,7 +55,8 @@ def readTestcase(self, filename):
             # Determine whether the f has a DDBEGIN..DDEND section.
             for line in f:
                 if line.find(b"DDEND") != -1:
-                    raise LithiumError("The testcase (%s) has a line containing 'DDEND' without a line containing 'DDBEGIN' before it." % self.filename)
+                    raise LithiumError("The testcase ({}) has a line containing 'DDEND' ".format(self.filename) +
+                                       "without a line containing 'DDBEGIN' before it.")
```

Comment:
This is marginally better like this:

    raise LithiumError("The testcase (%s) has a line containing 'DDEND'"
                       " without a line containing 'DDBEGIN' before it." % self.filename)

because the parser concatenates the string constants for you instead of doing it at runtime with `+`.

`%` or `.format()` still works even though the format placeholder is in the first string because this is a supported way of breaking string constants across lines.

I'm also against switching to `.format()` because it is more verbose, doesn't provide any value for our usage, and it's slower ([which is admitted in the python docs](https://docs.python.org/3/library/stdtypes.html#string-methods)). `%` formatting is not and probably never will be deprecated, and the "[quirks](https://docs.python.org/3/library/stdtypes.html#old-string-formatting)" listed in the docs are not very common and easily worked around. However if we do switch we should switch all at once (including loggers), not intermix styles.